### ISSUE TITLE
User request: Play through interactive studies sequentially, not randomly.

### ISSUE THREAD

**Issue by Siderite** [Siderite/lichessTools#416]
https://lichess.org/forum/team-l1chess-tools-users-team/request-for-interactive-studies-play-through-all-variations-sequentially

**Comment by Siderite** [comment]
Questions:
- If a variation has 10 moves and you play 9, does it count?
- Should the "score" have any relevance?
- Can you reset the stats so you start over and how?
- If you switch devices do you expect to have the playing stats transferred?
- Do you expect some visual confirmation of which variation you are playing and which you have played already?
- by sequentially do you mean you want to play them in order, or is it OK to be random?
- should played variations be erased if enough time passes?
- what do you expect to happen when you finish all variations?
- what happens if you are modifying the PGN?

Also remember that this will apply to all studies, not only yours.

**Comment by Siderite** [comment]
More discussion:

Once we finish to review a line , then the program ends, and we have to manually press a button in order to review the next variation of the game : there should be 2 improvements there :
- A more explicit button that would say : "review next variation", because at the moment, it's not clear on which button to pres. Personaly i press on the "return button" next to preview button, but i think there should be a better title/button, that people would understand much better 
- The second option which would be "much better" , would be the possibility that when we finish to review the Line A (after 10.Qf3), the program ITSELF goes back to 10.Qf3 and ask us for the SECOND variation there. The idea would be to do it AUTOMATICALLY , without asking us to push a button "next variation". do you see what I mean ?  It would be much more effective.
- Another idea would be like on chessable, after finishing one line, if a mistake was done while reviewing the line, to do the position again : so the program would go directly to where the mistake was done, and if we do the good move, it would go then to next mistake directly.  For learning purpose it would be great 

To summarize, here is what is missing : 
- possibility to continue to review directly from where we started (as written in my last mail), let's say at 10.Qf3
- Possibility to review Automatically (without confirming manually that we want to go next when we arrive at the end of a variation)
- Have a better explaining button for those who don't want to review Automatically at the end of a variation
- Possibility to come back only to where we went wrong, once we finish the line, before to go to next variation. Or At the end of the whole review

**Comment by Siderite** [comment]
This is already working since v2.3.20, but it is work in progress.

**Comment by Siderite** [comment]
I will close this and open new ones if requirements change.
